src/OrbitQt/CaptureOptionsDialog.cpp

```diff
@@ -194,6 +225,40 @@ DynamicInstrumentationMethod CaptureOptionsDialog::GetDynamicInstrumentationMeth
   ORBIT_UNREACHABLE();
 }
 
+void CaptureOptionsDialog::SetEnableCallStackCollectionOnThreadStateChanges(bool check) {
+  ui_->threadStateChangeCallstackCollectionCheckBox->setChecked(check);
+}
+
+bool CaptureOptionsDialog::GetEnableCallStackCollectionOnThreadStateChanges() const {
+  return ui_->threadStateChangeCallstackCollectionCheckBox->isChecked();
+}
+
+void CaptureOptionsDialog::SetThreadStateChangeCallstackMethod(
+    CaptureOptions::UnwindingMethod thread_state_change_callstack_method) {
+  switch (thread_state_change_callstack_method) {
+    case CaptureOptions::kFramePointers:
+      ui_->threadStateChangeCallstackFramepointersMethodRadioButton->setChecked(true);
+      ui_->threadStateChangeCallstackDWARFMethodRadioButton->setChecked(false);
+      break;
+    case CaptureOptions::kDwarf:
+    default:
+      ui_->threadStateChangeCallstackDWARFMethodRadioButton->setChecked(true);
+      ui_->threadStateChangeCallstackFramepointersMethodRadioButton->setChecked(false);
+      break;
+  }
+}
+
+orbit_grpc_protos::CaptureOptions::UnwindingMethod
+CaptureOptionsDialog::GetThreadStateChangeCallstackMethod() const {
+  if (ui_->threadStateChangeCallstackFramepointersMethodRadioButton->isChecked()) {
+    return CaptureOptions::kFramePointers;
+  }
+  if (ui_->threadStateChangeCallstackDWARFMethodRadioButton->isChecked()) {
+    return CaptureOptions::kDwarf;
+  }
+  return CaptureOptions::kDwarf;
```

Comment:
This is strange. I will rather return CaptureOptions::kUndefined;

Reply:
Done.